File: java/tests/StrataInspectorTest/src/strata1/inspector/diagnostic/DiagnosticCheckTest.java
// ##########################################################################
// # File Name:		DiagnosticCheckTest.java
// # Copyright(C):	2007, Capital Group Companies, Inc. 
// #                All Rights Reserved.
// ##########################################################################

package strata1.inspector.diagnostic;

import static org.junit.Assert.assertEquals;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;



/**
 * 
 * @author 		
 *     AFS Strategic Initiative 5 
 * @conventions	
 *     <a href="{@docRoot}/NamingConventions.html">Naming Conventions</a>
 */
public class DiagnosticCheckTest
{
	private DiagnosticCheck     itsTarget;
	private static final String NAME = "XYZ";

	/************************************************************************
	 *  
	 *
	 * @throws java.lang.Exception
	 */
	@Before
	public void setUp() throws Exception
	{
		itsTarget = new MockDiagnosticCheck( NAME,true,true,true,false );
	}

	/************************************************************************
	 *  
	 *
	 * @throws java.lang.Exception
	 */
	@After
	public void tearDown() throws Exception
	{
		itsTarget = null;
	}

	/**
	 * Test method for 
	 * {@link DiagnosticCheck#DiagnosticCheck(String)}.
	 */
	@Test
	public void testDiagnosticCheck()
	{
		assertEquals( 
		    NAME,
		    new MockDiagnosticCheck( NAME,true,true,true,true ).getName() );
	}

	/**
	 * Test method for 
	 * {@link DiagnosticCheck#runDiagnostic(IDiagnosticResult)}.
	 */
	@Test
	public void testRunDiagnostic()
	{
		MockDiagnosticResult   result   = new MockDiagnosticResult();
		MockDiagnosticReporter reporter = new MockDiagnosticReporter();
		
		result.attachReporter( reporter );
		itsTarget.runDiagnostic( result );
		result.verify();
	}

}


// ##########################################################################

File: legacy/csharp/tests/StrataAdministrationTest/Diagnostic/DiagnosticCheckTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;

namespace Strata.Administration.Diagnostic
{
    [TestClass]
    public 
    class DiagnosticCheckTest
    {
	    private DiagnosticCheck target;
	    private static String   NAME = "XYZ";

        [TestInitialize]
	    public void 
        SetUp()
	    {
		    target = new MockDiagnosticCheck( NAME,true,true,true,false );
	    }

        [TestCleanup]
	    public void 
        TearDown()
	    {
		    target = null;
	    }

        [TestMethod]
	    public void 
        TestDiagnosticCheck()
	    {
		    Assert.AreEqual( 
		        NAME,
		        new MockDiagnosticCheck( NAME,true,true,true,true ).Name );
	    }

        [TestMethod]
	    public void 
        TestRunDiagnostic()
	    {
		    MockDiagnosticResult result   = new MockDiagnosticResult();
		    IDiagnosticReporter  reporter = new Mock<IDiagnosticReporter>().Object;
		
		    result.AttachReporter( reporter );
		    target.RunDiagnostic( result );
		    Assert.IsTrue( result.ActualRun.Contains( NAME )  );
	    }
    }
}
